dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -626,6 +627,97 @@ class TypedDataset[T] protected[frameless](val dataset: Dataset[T])(implicit val
 
     TypedDataset.create[Out](selected)
   }
+
+  /**
+    *
+    * @see [[frameless.TypedDataset.withColumnApply#apply]]
+    * @tparam U the output type
+    */
+  def withColumnTyped[U] = new withColumnApply[U]
```

Comment:
Is scalac happy if you call this `withColumn`? I wouldn't mind using the same name...

Reply:
sadly it's not. `withColumn` would be the obvious choice. 2nd hardest thing in CS. I'm open to naming suggestions

Reply:
It might actually be better reversed: `withColumn` & `withColumnTupled`?

Reply:
Oh interesting. I think that makes sense, though returning tuples does seem to be the first class returned result in ex `select` etc. 
I wonder if the same should be done there or if a different first class result (eg `Record`) should be chosen.